include/swift/Basic/JSONSerialization.h

```diff
@@ -341,6 +342,15 @@ struct unvalidatedObjectTraits : public std::integral_constant<bool,
 && !has_ObjectValidateTraits<T>::value> {};
 
 class Output {
+public:
+  /// Enum of all Output subclasses to dynamically cast the output object and
+  /// thus pass serialization options
+  enum OutputKind {
+    Normal,
+    SyntaxTree
```

Comment:
Can we have some documentation here for the differences? a.k.a. why do we need node id in one case however not in the other.

Reply:
I added some documentation. I don't want to go into detail on how to the subclasses here because it's not really the point where I'd look for it.